concurrent_map.go

```diff
@@ -302,6 +348,40 @@ func (m ConcurrentMap) MarshalJSON() ([]byte, error) {
 	return json.Marshal(tmp)
 }
 
+// isEvictionOccurred check if the map size overflow the maximum allowed size.
+func (m ConcurrentMap) isEvictionOccurred(key string) bool {
+	shard := m.GetShard(key)
+	if m.Count() > shard.maxSize {
+		return true
+	}
+	return false
+}
+
+// removeFirstElement removes the first element from the key slice.
+func (m ConcurrentMap) removeFirstElement(key string) {
+	shard := m.GetShard(key)
+	shard.Lock()
+	copy(shard.mapKeys[0:], shard.mapKeys[1:])
```

Comment:
If we have a shard with 1000 elements for a total of 100k elements in the sharded map, then for every eviction we need to copy 999 elements. 
I am not sure if copy is smart enough to figure out this is a copy over itself and do optimizations.
So I would not do this through copying.

Reply:
Check my comment in your other pull request on elrond-go-sandbox

" One additional idea: For eviction instead of copying the slice of keys, you could just use an index that moves in a circular way over the array ( idx = (idx+1)% size ) and just replace the value pointed by the index with the new one - for a FIFO eviction scheme. This should be much cheaper. "

Reply:
I'm not sure if it is cheaper or not, but I did not observed any speed improvement using the suggested technique, since you have to iterate over each slice element and modify the value of the index the pointer is pointing too. Internally - from my perspective - the built in function is doing the same thing only in more optimized way, since it moves only the pointer around the slice elements. Also the on go slice tricks page it is suggested this way.

Anyway i have implemented as you have requested.

Reply:
It was probably not very clear what I meant, so I have created another branch with some quick changes, starting from your branch here: feat/EN-1525-cmap-with-eviction

maybe you can take a look, I did see some improvements running the benchmarks.

Reply:
Are you sure the branch exists? Git could not find it.

Reply:
maybe run git fetch first? github shows it if you check the active branches

Reply:
Sorry, i was looking in the main repo. 

I have checked your implementation, but please take care the ccache should implement the `RemoveOldest` method defined in the `Cacher` interface. This was one of the reason I have mentioned in one of the comments, that the new implementation should have to be adapted to the cacher interface.